pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1014,15 +1032,18 @@ func appNetworkDoActivateUnderlayNetwork(
 		}
 	}
 
-	bridgeIPAddr, appIPAddr, err := getUlAddrs(ctx, ulNum-1,
-		status.AppNum, ulStatus, netInstStatus)
+	appID := status.UUIDandVersion.UUID
+	appIPAddr, err := getUlAddrs(ctx, ulNum-1, netInstStatus, ulStatus,
+		appID)
 	if err != nil {
-		err := fmt.Errorf("Bridge/App IP address allocation failed: %v",
-			err)
+		err := fmt.Errorf("App IP address allocation failed: %v", err)
 		log.Error(err.Error())
 		addError(ctx, status, "getUlAddrs", err)
+		networkID := netInstStatus.UUID
+		appNumOnUNetFree(ctx, networkID, appID, true)
```

Comment:
This is broken since the Allocate and Free might not line up resulting in a Fatal in Free,
You need to call appNumOnUNetAllocate() in one and only one place and I think that needs to be a loop over config.UnderlayNetworkList at the top of appNetworkDoActivateAllUnderlayNetworks() so that they are allocated for all the underlays even if the activate fails.